src/applications/appeals/10182/tests/utils/redirect.unit.spec.js

```diff
@@ -0,0 +1,156 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { checkRedirect, onFormLoaded } from '../../utils/redirect';
+import { SHOW_PART3, SHOW_PART3_REDIRECT } from '../../constants';
+
+const pagesBeforePart3 = [
+  '/veteran-details',
+  '/homeless',
+  '/contact-information', // don't need to include edit pages, because they don't have a SIP link.
+  '/filing-deadlines',
+];
+
+const pagesAfter = ['/contestable-issues', '/review-and-submit'];
+const redirectPage = '/extension-request';
+
+describe('checkRedirect', () => {
+  it('should return false for pages before part 3 questions', () => {
+    expect(checkRedirect(pagesBeforePart3[0])).to.be.false;
+    expect(checkRedirect(pagesBeforePart3[1])).to.be.false;
+    expect(checkRedirect(pagesBeforePart3[2])).to.be.false;
+    expect(checkRedirect(pagesBeforePart3[3])).to.be.false;
+  });
+  it('should return true for pages not before part 3 questions', () => {
+    expect(checkRedirect(pagesAfter[0])).to.be.true;
+    expect(checkRedirect(pagesAfter[1])).to.be.true;
+  });
+});
+
+describe('onFormLoaded', () => {
+  const getProps = ({
+    showPart3 = true,
+    redirect = 'not-needed',
+    returnUrl,
+    pushSpy = () => {},
+  }) => ({
+    formData: {
+      [SHOW_PART3]: showPart3,
+      [SHOW_PART3_REDIRECT]: redirect,
+    },
+    returnUrl,
+    routes: [
+      {
+        pageList: [
+          {
+            pageKey: 'vetInfo',
+            path: pagesBeforePart3[0],
+          },
+          {
+            pageKey: 'homeless',
+            path: pagesBeforePart3[1],
+          },
+          {
+            pageKey: 'contactInfo',
+            path: pagesBeforePart3[2],
+          },
+          {
+            pageKey: 'deadlines',
+            path: pagesBeforePart3[3],
+          },
+          {
+            pageKey: 'extension request',
+            path: showPart3 ? redirectPage : '/hidden',
+          },
+          {
+            pageKey: 'contestable issues',
+            path: pagesAfter[0],
+          },
+          {
+            pageKey: 'other',
+            path: pagesAfter[1],
+          },
+        ],
+      },
+    ],
+    router: {
+      push: pushSpy,
+    },
+  });
+
+  describe('redirect not-needed flag', () => {
+    it('should not redirect from contact info page if showPart3 is true', () => {
+      const pushSpy = sinon.spy();
+      const props = getProps({ pushSpy, returnUrl: pagesBeforePart3[2] });
+      onFormLoaded(props);
+      expect(pushSpy.args[0][0]).to.eq(pagesBeforePart3[2]);
+    });
+    it('should not redirect from deadlines page if showPart3 is false', () => {
+      const pushSpy = sinon.spy();
+      const props = getProps({
+        pushSpy,
+        showPart3: false,
```

Comment:
is there a reason this test has `showPart3` as `false` when the rest of the tests in the describe have `showPart3` as `true`?

Reply:
The wrappers are grouping the redirect flag, so I wanted 2 tests in each group to check the `showPart3` flag in both states. The `onFormLoaded` callback depends on both the `SHOW_PART3` feature flag and the `SHOW_PART3_REDIRECT` values